src/components/Radio/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { RowBetween } from 'components/Row'
+import { darken } from 'polished'
+import { PropsWithChildren } from 'react'
+import styled, { DefaultTheme } from 'styled-components/macro'
+
+const Button = styled.button<{ isActive?: boolean; activeElement?: boolean }>`
+  align-items: center;
+  background: transparent;
+  border: 2px solid ${({ theme, isActive }) => (isActive ? theme.accentAction : theme.backgroundOutline)};
+  border-radius: 50%;
+  cursor: pointer;
+  display: flex;
+  outline: none;
+  padding: 5px;
+  width: fit-content;
+`
+
+const ToggleElementHoverStyle = (hasBgColor: boolean, theme: DefaultTheme, isActive?: boolean) =>
+  hasBgColor
+    ? {
+        opacity: '0.8',
+      }
+    : {
+        background: isActive ? darken(0.05, theme.accentAction) : darken(0.05, theme.deprecated_bg4),
+        color: isActive ? theme.white : theme.textTertiary,
+      }
+
+const ButtonFill = styled.span<{ isActive?: boolean; bgColor?: string }>`
+  animation: 0.1s opacity ease-in;
```

Comment:
You are right, there are constants, I will re-use.

Reply:
For some reason, I am having a hard time debugging this animation. I will remove this from the PR, so that we can ship it, and follow-up immediately to do this and Expand animation.